imports/ui/pages/resources/components/ResourceForm.jsx

```diff
@@ -0,0 +1,288 @@
+import React, { useState, useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import {
+  Box,
+  Button,
+  Center,
+  Flex,
+  FormControl,
+  FormLabel,
+  Input,
+  Switch,
+  Text,
+  VStack,
+  Wrap,
+  IconButton,
+  WrapItem,
+} from '@chakra-ui/react';
+import { SmallCloseIcon } from '@chakra-ui/icons';
+import ReactQuill from 'react-quill';
+import { sortableContainer, sortableElement } from 'react-sortable-hoc';
+import arrayMove from 'array-move';
+import AutoCompleteSelect from 'react-select';
+import makeAnimated from 'react-select/animated';
+
+import { editorFormats, editorModules } from '../../../@/constants/quillConfig';
+import { call, resizeImage, uploadImage } from '../../../@/shared';
+import { message } from '../../../components/message';
+import FormField from '../../../components/FormField';
+import FileDropper from '../../../components/FileDropper';
+import NiceSlider from '../../../components/NiceSlider';
+import Loader from '../../../components/Loader';
+
+const animatedComponents = makeAnimated();
+
+function ResourceForm({ defaultValues, isEditMode, history }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [resourceLabels, setResourceLabels] = useState([]);
+  const [resourcesForCombo, setResourcesForCombo] = useState([]);
+  const defaultImages = defaultValues?.images ? defaultValues.images : [];
+  const [images, setImages] = useState(defaultImages);
+
+  const { formState, handleSubmit, getValues, register, control } = useForm({
+    defaultValues,
+  });
+  const { isDirty, isSubmitting } = formState;
+  const isCombo = getValues('isCombo');
+
+  const [t] = useTranslation('resources');
+  const [tc] = useTranslation('common');
+
+  useEffect(() => {
+    getResourceLabels();
```

Comment:
`getResources` instead?

Reply:
Yeah getResources does nearly the same. I did a seperate method cause form component doesnt need anything more than id or label of resources so I made a custom method for selectbox data only instead of calling all resources data method. But ofcource we dont have to use that way.